public/app/features/dashboard-scene/scene/AddLibraryPanelDrawer.tsx

```diff
@@ -53,22 +39,9 @@ export class AddLibraryPanelDrawer extends SceneObjectBase<AddLibraryPanelDrawer
         throw new Error('Trying to replace a panel that does not have a DashboardGridItem');
       }
 
-      gridItemToReplace.setState({ body });
+      gridItemToReplace.setState({ body: newPanel });
     } else {
-      const newGridItem = new DashboardGridItem({
-        height: NEW_PANEL_HEIGHT,
-        width: NEW_PANEL_WIDTH,
-        x: 0,
-        y: 0,
-        body: body,
-        key: `grid-item-${panelId}`,
-      });
-
-      layout.setState({ children: [newGridItem, ...layout.state.children] });
-
-      if (!dashboard.state.isEditing) {
-        dashboard.onEnterEditMode();
-      }
+      dashboard.addPanel(newPanel);
```

Comment:
Same here when it comes to my previous comment about not duplicating interfaces. In this case it's slightly different tho as `addPanel` has also edit mode check, so seems we gonna have this method anyways. Wouldn't think of adding isEditing state to the layout manager.